core/src/com/unciv/logic/battle/BattleDamage.kt

```diff
@@ -136,11 +136,11 @@ object BattleDamage {
                 modifiers["Landing"] = -50
 
             // Land Melee Unit attacking to Water
-            if (!attacker.unit.isEmbarked() && attacker.isMelee() && defender.getTile().isWater
+            if (attacker.unit.type.isLandUnit() && !attacker.unit.isEmbarked() && attacker.isMelee() && defender.getTile().isWater
                     && !attacker.unit.hasUnique(UniqueType.AttackAcrossCoast))
                 modifiers["Boarding"] = -50
             // Naval Unit Melee attacking to Land (not City) unit
-            if (attacker.unit.type.isWaterUnit() && attacker.isMelee() && !defender.getTile().isWater
+            if (attacker.unit.type.isWaterUnit() && attacker.isMelee() && !defender.getTile().isLand
```

Comment:
I'm reasonably sure this was already the right way around

Reply:
You're right. Need to get some time to actually test